hal/us_ticker_api.h

```diff
@@ -47,39 +47,123 @@ void us_ticker_irq_handler(void);
 
 /** Initialize the ticker
  *
+ * Initialize or re-initialize the ticker. This resets all the
+ * clocking and prescaler registers, along with disabling
+ * the compare interrupt.
+ *
+ * @note Initialization properties tested by ::ticker_init_test
+ *
+ * Pseudo Code:
+ * @code
+ * void us_ticker_init()
+ * {
+ *     // Enable clock gate so processor can read TIMER registers
+ *     POWER_CTRL |= POWER_CTRL_TIMER_Msk;
+ *
+ *     // Disable the timer and ensure it is powered down
+ *     TIMER_CTRL &= ~(TIMER_CTRL_ENABLE_Msk | TIMER_CTRL_COMPARE_ENABLE_Msk);
+ *
+ *     // Configure divisors
+ *     uint32_t prescale = SystemCoreClock / 1000000;
+ *     TIMER_PRESCALE = prescale - 1;
+ *     TIMER_CTRL |= TIMER_CTRL_ENABLE_Msk;
+ *
+ *     // Install the interrupt handler
+ *     NVIC_SetVector(TIMER_IRQn, (uint32_t)us_ticker_irq_handler);
+ *     NVIC_EnableIRQ(TIMER_IRQn);
+ * }
+ * @endcode
  */
 void us_ticker_init(void);
 
 /** Read the current counter
```

Comment:
Does us_ticker_read() function returns ticker ticks or microseconds? It looks like that in previous implementation it was assumed that ticker clock speed is 1 MHz, so 1 tick is equal to 1 us, but the concept has changed, and clock speed might be different now. This means that description of the return value should not mention about microseconds, but ticker ticks count instead. Also function name is misleading now.

Reply:
Its returns ticks. I'll fix the doxygen to reflect this.

I agree, the name is misleading. Only potential solution to this is to rename the microsecond ticker (us_ticker_*) to high frequency ticker (hf_ticker_*). Another possible solution is to require the us ticker has a fixed 1MHz frequency.

Reply:
It looks like the better solution is to rename the function taking into account new features provided in this PR.